src/genCid.js

```diff
@@ -4,29 +4,38 @@ const CID = require('cids')
 const multicodec = require('multicodec')
 const multihashing = require('multihashing-async')
 
-exports = module.exports
+const codec = multicodec.DAG_PB
+const defaultHashAlg = multicodec.SHA2_256
 
-exports.codec = multicodec.DAG_PB
-exports.defaultHashAlg = multicodec.SHA2_256
+/**
+ * @typedef {object} GenCIDOptions - Options to create the CID
+ * @property {CID.CIDVersion} [cidVersion=1] - CID version number
+ * @property {multicodec.CodecNumber} [hashAlg=multicodec.SHA2_256] - Defaults to the defaultHashAlg of the format
+ */
 
 /**
  * Calculate the CID of the binary blob.
  *
- * @param {Object} binaryBlob - Encoded IPLD Node
- * @param {Object} [userOptions] - Options to create the CID
- * @param {number} [userOptions.cidVersion=1] - CID version number
- * @param {string} [UserOptions.hashAlg] - Defaults to the defaultHashAlg of the format
- * @returns {Promise.<CID>}
+ * @param {Uint8Array} binaryBlob - Encoded IPLD Node
+ * @param {GenCIDOptions} [userOptions] - Options to create the CID
  */
-const cid = async (binaryBlob, userOptions) => {
-  const defaultOptions = { cidVersion: 1, hashAlg: exports.defaultHashAlg }
-  const options = Object.assign(defaultOptions, userOptions)
+const cid = async (binaryBlob, userOptions = {}) => {
+  const options = {
+    cidVersion: userOptions.cidVersion == null ? 1 : userOptions.cidVersion,
+    hashAlg: userOptions.hashAlg || defaultHashAlg
```

Comment:
That would break if you use the identity hash `0x00`.

Reply:
Fixed